File: backend/tests/test_data/open_validation/benchmark_logistic.R
# benchmark_logistic.R
# Insight Platform - Open Validation Pack
# Goal: Verify Logistic Regression Coefficients

# 1. Load Data
df <- read.csv("../benchmark_logistic.csv")

# 2. Fit GLM (Logistic)
model <- glm(grade ~ gpa + tuce + psi, data = df, family = binomial(link = "logit"))

# 3. Output Results
print(summary(model))
print("Coefficients:")
print(coef(model))
print("ConfInt:")
print(confint.default(model))

File: backend/tests/test_data/open_validation/benchmark_logistic.py
import pandas as pd
import statsmodels.formula.api as smf

# 1. Load Data
df = pd.read_csv("../benchmark_logistic.csv")

# 2. Fit Logistic Regression
model = smf.logit("grade ~ gpa + tuce + psi", data=df).fit()

# 3. Output Results
print(model.summary())
print("Coefficients:")
print(model.params)
print("Conf Int:")
print(model.conf_int())
